apps/api/src/donations/donations.service.ts

```diff
@@ -114,6 +115,87 @@ export class DonationsService {
     return donation
   }
 
+  /**
+   * Create initial donation object for tracking purposes
+   * This is used when the payment is created from the payment intent
+   */
+  async createInitialDonationFromIntent(
+    campaign: Campaign,
+    stripePaymentDto: CreateStripePaymentDto,
+    paymentIntent: Stripe.PaymentIntent,
+  ): Promise<Donation> {
+    Logger.debug('[ CreateInitialDonationFromIntent]', {
+      campaignId: campaign.id,
+      amount: paymentIntent.amount,
+      paymentIntentId: paymentIntent.id,
+    })
+
+    /**
+     * Create or connect campaign vault
+     */
+    const vault = await this.prisma.vault.findFirst({ where: { campaignId: campaign.id } })
+    const targetVaultData = vault
+      ? // Connect the existing vault to this donation
+        { connect: { id: vault.id } }
+      : // Create new vault for the campaign
+        { create: { campaignId: campaign.id, currency: campaign.currency, name: campaign.title } }
+
+    /**
+     * Create or update initial donation object
+     */
+    const donation = await this.prisma.donation.upsert({
+      where: { extPaymentIntentId: paymentIntent.id },
+      create: {
+        amount: 0, //this will be updated on successful payment event
+        chargedAmount: paymentIntent.amount,
+        currency: campaign.currency,
+        provider: PaymentProvider.stripe,
+        type: DonationType.donation,
+        status: DonationStatus.initial,
+        extCustomerId: stripePaymentDto.personEmail ?? '',
+        extPaymentIntentId: paymentIntent.id,
+        extPaymentMethodId: 'card',
+        billingEmail: stripePaymentDto.isAnonymous ? stripePaymentDto.personEmail : null, //set the personal mail to billing which is not public field
```

Comment:
Maybe we can add the billing email in all cases, not only whey user has selected anonymous donation.

Reply:
That would make sense, it would actually make more sense to not save the email if he has chosen `isAnonymous` if it is not required.
Then we might not be able to send the receipt though if this is the email that is used for that.

Reply:
As you said the `billingEmail` is never exposed to the public, so it's always anonymous.

I'd suggest we keep the billing email so we can contact the person if there's a problem with the payments or other circumstances.

Reply:
[407 - always add billingEmail as the provided personEmail](https://github.com/podkrepi-bg/api/pull/425/commits/7db3a5a62923e26b4a3a0c38b48ede847cc62eb3)